forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabRepository.java

```diff
@@ -286,21 +286,34 @@ public VCS repositoryType() {
     }
 
     @Override
-    public String fileContents(String filename, String ref) {
-        var confName = URLEncoder.encode(filename, StandardCharsets.UTF_8);
-        var conf = request.get("repository/files/" + confName)
-                          .param("ref", ref)
-                          .onError(response -> {
-                              log.warning("First time request returned bad status: " + response.statusCode());
-                              log.info("First time response body: " + response.body());
-                              // Retry once with additional escaping of the path fragment
-                              var escapedConfName = URLEncoder.encode(confName, StandardCharsets.UTF_8);
-                              return Optional.of(request.get("repository/files/" + escapedConfName)
-                                            .param("ref", ref).execute());
-                          })
-                          .execute();
-        var content = Base64.getDecoder().decode(conf.get("content").asString());
-        return new String(content, StandardCharsets.UTF_8);
+    public Optional<String> fileContents(String filename, String ref) {
+        var encodedFileName = URLEncoder.encode(filename, StandardCharsets.UTF_8);
+        JSONValue content;
+        try {
+            content = request.get("repository/files/" + encodedFileName)
+                    .param("ref", ref)
+                    .onError(response -> {
+                        // Retry once with additional escaping of the path fragment
+                        // Only retry when the error is exactly "File Not Found"
+                        if (response.statusCode() == 404 && response.body().contains("File Not Found")) {
```

Comment:
The comment says 'exactly "File Not Found"' but the conditional uses `.contains`. I think we can switch to `.equals`.

Reply:
I use contains because the real body is `{"message":"404 File Not Found"}`.  I will use  `response.body().equals("{\"message\":\"404 File Not Found\"}")`